### PR TITLE
fix(tmux): resolve exit-detection conflation in IsAlive — closes P1 from #2831

### LINKED ISSUE #2802: bug(runtime/tmux): IsAlive reports pane-exists as agent-alive, masking crashed agents behind the keep-alive shell
## Problem

`tmux.Runtime.IsAlive` (`backend/internal/adapters/runtime/tmux/tmux.go`) only
checks whether the tmux **session** exists (`tmux has-session`). It has no way
to tell whether the **agent process** inside the pane is still running.

The launch command built by `buildLaunchCommand` always appends a keep-alive
shell after the agent's argv:

```go
b.WriteString(strings.Join(parts, " "))
// Keep the tmux session alive after the agent exits so the operator can
// inspect the terminal.
b.WriteString(`; exec "${SHELL:-/bin/sh}" -i`)
```

`;` runs unconditionally regardless of exit status, so a crash, an OOM kill,
or a clean exit all leave the pane alive on an interactive shell. From that
point on, `has-session` returns true forever, `IsAlive` reports `true`
forever, and the reaper (`backend/internal/observe/reaper/reaper.go`,
`probeOne`) only ever reports `ProbeAlive` to the lifecycle manager — it has
no signal that the agent itself is gone.

## Impact

- A session whose agent crashes while in a sticky state (`blocked` /
  `waiting_input`) never gets a `ProbeDead` fact, so the lifecycle fix in
  #2749 (for #2745) can't trigger for the tmux runtime — the sticky-state
  reducer fix is correct, but the fact it depends on is never produced for
  most real crashes. See [@neversettle17-101's review comment on
  #2749](https://github.com/AgentWrapper/agent-orchestrator/pull/2749#pullrequestreview-4729018515).
- Same architectural gap as #2089 (filed against the old TypeScript
  implementation's `isAlive()`); it was never fixed for the Go rewrite's
  tmux adapter.
- Any other automation that trusts `IsAlive`/`ProbeAlive` (session restore,
  dashboard status) is misled the same way.

## Repro

1. Spawn a session on the tmux runtime.
2. Find the agent's PID inside the pane and `kill -9` it (or let it crash),
   leaving the pane's shell alive.
3. Call `IsAlive` (or wait for the reaper's next tick) — it still reports
   `true` / `ProbeAlive`, even though the agent is gone.

## Proposed fix

Keep the fix contained to the tmux adapter — no changes needed to
`ports.ProbeResult`, `RuntimeFacts`, the reaper, or lifecycle.

1. In `buildLaunchCommand`, mark the pane via tmux's own per-pane user
   option right before the keep-alive exec:

   ```
   <agent argv...>; tmux set-option -p -t "$TMUX_PANE" @ao_agent_exited 1 2>/dev/null; exec "${SHELL:-/bin/sh}" -i
   ```

2. In `IsAlive`, after `has-session` succeeds, additionally query that pane
   option (`tmux show-options -v -p -t <id> @ao_agent_exited`). If it reads
   `1`, the agent has exited even though the pane survives — return
   `false`, not `true`.

This makes a confirmed agent exit surface as `ProbeDead` through the
existing reaper/lifecycle path unchanged, so #2749's reducer fix actually
fires for the case it was written for.

## Test plan (TDD)

In `backend/internal/adapters/runtime/tmux/tmux_test.go`:

1. `TestRuntime_IsAlive_ReportsDeadWhenAgentExitedBehindSurvivingPane` —
   `has-session` succeeds, pane option reads `1` → `IsAlive` returns
   `false, nil`. Currently fails (returns `true`).
2. `TestRuntime_IsAlive_ReportsAliveWhenAgentStillRunning` — `has-session`
   succeeds, pane option unset/empty → `IsAlive` returns `true, nil`.
   Should pass (regression guard).
3. `TestRuntime_IsAlive_SessionMissingStillReportsDead` — existing
   `has-session`-missing path unchanged.
4. `TestRuntime_IsAlive_PaneOptionProbeErrorReportsFailedProbe` — the
   `show-options` call itself errors (not "option unset") → surfaced as a
   probe error, never collapsed to alive or dead.

## Related issues

- #2745 (sticky-state zombie sessions) / #2749 (its fix, pending review)
- #2089 (same gap, old TS implementation)

### PR DESCRIPTION
**Pull request by anmol0b**
## What
Fixes IsAlive in the tmux runtime adapter so it no longer conflates "session doesn't exist" with "session exists but the agent has exited behind a surviving keep-alive pane" — both previously returned (false, nil). IsAlive now self-heals by destroying the session once it confirms the agent has exited, so false always means "nothing left to clean up." Also fixes Create's post-launch readiness check to use a plain session-existence check instead of IsAlive, avoiding a false positive on fast-exiting agents.
Brief description of the change.

## Why
Fixes #2802, which I opened after tracing a session-collision bug back to IsAlive conflating "session doesn't exist" with "session exists but the agent exited behind a surviving keep-alive pane" — both returned (false, nil).

The reaper treats IsAlive == false as "already gone, nothing to reap." That's correct when the session never existed, but wrong when the agent died and the pane survived — the session was never torn down, and a later Restore/Create with the same session ID collided with the still-existing tmux session. @nikhilachale flagged the same root cause as a P1 blocking #2831, still unresolved there.

## How
- Split IsAlive into hasSession (session existence) and agentExited (pane-local @ao_agent_exited marker check), reused by both Create and IsAlive instead of duplicating the has-session call.
- Create's readiness check now calls hasSession instead of IsAlive, so a quick command that exits before Create finishes its check doesn't get misreported as "exited before ready."
- IsAlive calls Destroy internally once it confirms the exit marker, before returning false, nil. Destroy is already idempotent, so this is safe regardless of caller behavior.
- A failed cleanup Destroy inside IsAlive does not get surfaced as a probe error — the agent is still confirmed dead even if best-effort cleanup didn't land. This intentionally trades a query method having a side effect for closing the collision at the source; flagged for reviewer discussion in case a tri-state result (alive / dead-clean / dead-needs-reap) is preferred long-term instead.
- Complementary to #2749 (lifecycle reducer), which depends on this adapter emitting a trustworthy ProbeDead/IsAlive signal in the first place.

## Testing

- `go build ./internal/adapters/runtime/tmux/...`
- `go test ./internal/adapters/runtime/tmux/... -v` — all existing tests pass, plus new:
  - `TestIsAlive_DestroysSessionWhenAgentExitedBehindSurvivingPane` — confirms `Destroy` (kill-session) fires when the exit marker is set.
  - `TestIsAlive_ReportsDeadEvenIfCleanupDestroyFails` — confirms a failed cleanup `Destroy` still returns `alive=false, err=nil`.
  - `TestCreate_SucceedsAfterPriorSessionMarkedExited` — documents that `Create` succeeds against a session ID previously cleaned up by `IsAlive`.

Screenshot 2026-07-21 at 11 41 50 PM

## Checklist

- [x] Branched from `main` (or continuing an existing PR branch)
- [x] One focused change; links the related issue when applicable
- [ ] Follows [AGENTS.md](https://github.com/AgentWrapper/agent-orchestrator/blob/main/AGENTS.md) conventions and PR hygiene
- [x] Tests added/updated for user-visible behavior where it makes sense
- [ ] Relevant CI checks pass for the area touched (`go`, frontend, etc.)

### CODE REVIEW

**Comment by somewherelostt** [comment]
Thanks for contributing to Agent Orchestrator.

This PR is being picked up by the current external contributor on-call pair:

- @illegalcall
- @pulkit7070

If someone is already working on this, please continue as usual.
The on-call pair is added for visibility, tracking, and support, not to take over the work.
If you need help with review, direction, reproduction, or next steps, please tag @illegalcall and @pulkit7070 here.

For faster context or live questions, you can also join the AO Discord.

Join the session here:
https://discord.gg/H6ZDcUXmq

Come by if you want to see what is being built, ask questions, or just hang around with the community.

**Review by Pulkit7070 — COMMENTED** [review]
Reviewed at head c1fc1db. Ran from a clean checkout:

- `go test ./internal/adapters/runtime/tmux -count=1` -> ok (1.18s)
- `go vet ./internal/adapters/runtime/tmux` -> clean
- `go build ./...` -> ok

The core fix is right. Splitting `hasSession` (pure existence) from `IsAlive` (agent liveness) removes the conflation where a surviving keep-alive shell made a dead agent look alive. Setting the `@ao_agent_exited` pane option with `;` (not `&&`) means it is set whether the agent exits 0 or non-zero, so a crashed agent is correctly reported dead. Using `show-options -q` so an unset option is exit-0-empty (alive / pre-marker session) while a real failure stays a non-zero probe error is the correct distinction, and `Create` now using `hasSession` rather than `IsAlive` is the right call since a freshly launched agent has no marker yet. Tests cover dead-behind-surviving-pane, unset-means-alive, and genuine-probe-failure. Nicely done.

Two edge cases worth considering (inline), neither blocking.

File: backend/internal/adapters/runtime/tmux/tmux.go [hunk 1]
```diff
@@ -553,6 +596,12 @@ func buildLaunchCommand(cfg ports.RuntimeConfig) string {
 		parts[i] = shellQuote(a)
 	}
 	b.WriteString(strings.Join(parts, " "))
+	// Mark the pane as agent-exited via tmux's own per-pane option before the
+	// keep-alive shell takes over. has-session alone cannot tell the agent
+	// process died from the pane merely surviving on this fallback shell, so
+	// IsAlive queries this option too (see agentExited). "2>/dev/null" keeps a
+	// set-option failure from surfacing in the interactive shell that follows.
+	b.WriteString(`; tmux set-option -p -t "$TMUX_PANE" ` + agentExitedOption + ` 1 2>/dev/null`)
```
Comment (Pulkit7070) [inline backend/internal/adapters/runtime/tmux/tmux.go:736]: set-option is written pane-scoped against `$TMUX_PANE`, but `agentExited` reads it via `show-options -p -t <sessionID>`, which resolves to the session's currently active pane. For the single-pane sessions this runtime creates that is the same pane. If an operator splits the window (this is an inspectable keep-alive shell), the active pane at probe time may differ from the agent's pane and the read could miss the marker, reporting a dead agent as alive. Consider targeting the specific pane id on read, or documenting the single-pane assumption.

Comment (Pulkit7070) [inline backend/internal/adapters/runtime/tmux/tmux.go:736]: `2>/dev/null` on set-option is reasonable to keep noise out of the interactive shell, but note it also swallows a transient set-option failure. Since this runs exactly once at agent exit, if that single set-option fails while the session survives, the marker is never written and IsAlive will report the dead agent as alive indefinitely (no retry). has-session catches the common case (server/session gone) but not a lone set-option failure with a live server. Narrow, but worth a comment acknowledging it.

**Comment by anmol0b** [comment]
@Pulkit7070 
Fixed : exitedOptionArgs now targets pane index .0 explicitly, since that's always the pane new-session creates and the one the marker-write targets via $TMUX_PANE. A later window split can no longer cause the read to resolve to a different, unmarked pane. Updated TestCommandBuilders to lock in the .0 suffix.

**Comment by anmol0b** [comment]
@Pulkit7070 
Added a comment acknowledging the gap ,  went with documenting rather than retrying, since the failure mode is narrow (set-option failing while tmux itself stays healthy) and has-session already covers the common case. Happy to add a retry if you'd rather close it fully.

**Comment by nikhilachale** [comment]
Closing this PR because of duplicacy
